typed-racket-lib/typed-racket/base-env/base-env-indexing-abs.rkt

```diff
@@ -3,12 +3,18 @@
 (require
  "../utils/utils.rkt"
  (for-template racket/base racket/list racket/unsafe/ops racket/flonum racket/extflonum racket/fixnum)
+ (for-syntax racket/base syntax/parse)
  (utils tc-utils) 
  (rename-in (types abbrev numeric-tower) [-Number N] [-Boolean B] [-Symbol Sym]))
 
 (provide indexing)
 
-(define-syntax-rule (make-env* [i t] ...) (make-env [i (λ () t)] ...))
+(define-syntax (make-env* stx)
+  (syntax-parse stx
+   [(_ [id ty (~optional trusted-positive? #:defaults ([trusted-positive? #'#true]))] ...)
+    #'(make-env [id ty trusted-positive?] ...)]))
+
+(define needs-transient-check #false)
```

Comment:
note: these env changes are a hack

Shallow TR usually changes `(f x)` to a `(check T (f x))` to spot-check the values that `f` returns. The env changes let us skip the check for certain identifiers `f`. Ideally we'd extend the language of types to make that decision ... but I think we should leave that for the future.

TODO write a comment somewhere that explains this